jsl-tinylog/src/main/java/org/tinylog/jsl/TinylogLogger.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2021 Gerrit Rode
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.tinylog.jsl;
+
+import java.util.Objects;
+import java.util.ResourceBundle;
+import java.util.function.Supplier;
+
+import org.tinylog.configuration.Configuration;
+import org.tinylog.format.JavaTextMessageFormatFormatter;
+import org.tinylog.format.MessageFormatter;
+import org.tinylog.provider.LoggingProvider;
+import org.tinylog.provider.ProviderRegistry;
+
+/**
+ * Java System.Logger compatible logger that uses tinylog's {@link LoggingProvider}.
+ */
+public class TinylogLogger implements System.Logger {
+
+	private static final int STACKTRACE_DEPTH = 2;
+
+	private static final MessageFormatter MESSAGE_FORMATTER = new JavaTextMessageFormatFormatter(Configuration.getLocale());
+	private static final LoggingProvider LOGGING_PROVIDER = ProviderRegistry.getLoggingProvider();
+
+	private final String name;
+	private final Module module;
+
+	/**
+	 * @param name   the name of the logger
+	 * @param module the module for which the logger is being created
+	 */
+	public TinylogLogger(final String name, final Module module) {
+		this.name = Objects.requireNonNull(name);
+		this.module = Objects.requireNonNull(module);
+	}
+
+	@Override
+	public String getName() {
+		return name;
+	}
+
+	@Override
+	public boolean isLoggable(final Level level) {
+		Objects.requireNonNull(level);
+
+		// Setting the log level of a message to OFF makes little sense and should result in the message not being logged.
+		if (level == Level.OFF) {
+			return false;
+		} else {
+			return LOGGING_PROVIDER.isEnabled(STACKTRACE_DEPTH, null, translate(level));
```

Comment:
Potential performance improvement: Store the minimum active severity level as `static final` like in [slf4j/TinylogLogger.java](https://github.com/tinylog-org/tinylog/blob/42c4b8e855dd765814ec2555f1b235ff34448e85/slf4j-tinylog/src/main/java/org/tinylog/slf4j/TinylogLogger.java). This allows the JVM to identify which logging methods will never output anything at runtime and eliminate these calls completely from the byte code (see conclusion of https://tinylog.org/v2/benchmark/ for details).

Reply:
Ok, I looked at the example. I just don't see how the approach can be implemented in System.Logger. In System.Logger there are no different methods for TRACE to ERROR. There is basically only one log method, which gets the log level passed to it. So I know only at runtime whether the passed level can be logged.

Reply:
Sorry! You are right :) All other logging API adapters provide separate methods for each severity level. Somehow I have assumed that System.Logger has such methods, too.